airbyte-integrations/bases/base-java/src/test/java/io/airbyte/integrations/destination_async/BufferManagerDequeueTest.java

```diff
@@ -0,0 +1,58 @@
+package io.airbyte.integrations.destination_async;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+
+import io.airbyte.commons.json.Jsons;
+import io.airbyte.integrations.destination_async.BufferManager.BufferManagerDequeue;
+import io.airbyte.integrations.destination_async.BufferManager.BufferManagerDequeue.Batch;
+import io.airbyte.integrations.destination_async.BufferManager.BufferManagerEnqueue;
+import io.airbyte.protocol.models.v0.AirbyteMessage;
+import io.airbyte.protocol.models.v0.AirbyteMessage.Type;
+import io.airbyte.protocol.models.v0.AirbyteRecordMessage;
+import io.airbyte.protocol.models.v0.StreamDescriptor;
+import org.junit.jupiter.api.Test;
+
+public class BufferManagerDequeueTest {
+  private static final String STREAM_NAME = "stream1";
+  private static final StreamDescriptor STREAM_DESC = new StreamDescriptor().withName(STREAM_NAME);
+  private static final String RECORD = "abc";
+  private static final AirbyteMessage RECORD_MSG = new AirbyteMessage()
+      .withType(Type.RECORD)
+      .withRecord(new AirbyteRecordMessage()
+          .withStream(STREAM_NAME)
+          .withData(Jsons.jsonNode(RECORD)));
+
+  @Test
+  void testReadFromBatch() {
+    final BufferManager bufferManager = new BufferManager();
+    final BufferManagerEnqueue enqueue = bufferManager.getBufferManagerEnqueue();
+    final BufferManagerDequeue dequeue = bufferManager.getBufferManagerDequeue();
+
+    enqueue.addRecord(STREAM_DESC, RECORD_MSG);
+    enqueue.addRecord(STREAM_DESC, RECORD_MSG);
+    enqueue.addRecord(STREAM_DESC, RECORD_MSG);
+    enqueue.addRecord(STREAM_DESC, RECORD_MSG);
+
+    try(final Batch take = dequeue.take(STREAM_DESC, 40)) {
+      assertEquals(2, take.getData().toList().size());
```

Comment:
why is this 2 and not 4?

Reply:
because the 4 records are 80 bytes but i only asked for 40 bytes of records.

Reply:
ah ok, not obvious abc = 20 bytes.

Reply:
i'll add a comment in the next PR.